src/Components/App/App.js

```diff
@@ -62,22 +68,74 @@ class App extends Component {
   }
 
   setActivities = (activities) => {
+    activities.forEach(activity => {
+      this.getActivityData(activity)
+    })
     this.setState({activities: [...activities]})
   }
 
   determineRandomActivity = () => {
+    let allFilteredActivities = this.state.activities.reduce((filtered, activity) => {
+      if (activity.includes('games')) {
+        activity = activity.replace(/ games/gi, 'Games')
+      }
+      let filteredActivities = this.filterActivity(activity, this.state[`${activity}Answers`])
+      filtered.push(filteredActivities)
+      return filtered
+    }, [])
+    let final = allFilteredActivities.flat()
+    let randomNumber = Math.floor(Math.random() *final.length)
+    return final[randomNumber]
+  }
```

Comment:
While it's not the prettiest function, It has some pretty solid logic regarding the filtering of the activities based on user responses.